src/Microsoft.Extensions.CodeGenerators.Mvc/Controller/ControllerWithContextGenerator.cs

```diff
@@ -26,8 +40,87 @@ public abstract class ControllerWithContextGenerator : ControllerGeneratorBase
+                AreaName = string.Empty, //ToDo
+                UseAsync = controllerGeneratorModel.UseAsync,
+                ControllerNamespace = GetControllerNamespace(),
+                ModelMetadata = modelTypeAndContextModel.ModelMetadata
+            };
+
+            await CodeGeneratorActionsService.AddFileFromTemplateAsync(outputPath, GetTemplateName(controllerGeneratorModel), TemplateFolders, templateModel);
+            Logger.LogMessage("Added Controller : " + outputPath.Substring(ApplicationEnvironment.ApplicationBasePath.Length));
+
+            await GenerateViewsIfRequired(controllerGeneratorModel, modelTypeAndContextModel, templateModel.ControllerRootName);
+        }
+
+        private async Task GenerateViewsIfRequired(CommandLineGeneratorModel controllerGeneratorModel,
+            ModelTypeAndContextModel modelTypeAndContextModel,
+            string controllerRootName)
+        {
+            if (!controllerGeneratorModel.IsRestController && !controllerGeneratorModel.NoViews)
+            {
+                var layoutDependencyInstaller = ActivatorUtilities.CreateInstance<MvcLayoutDependencyInstaller>(ServiceProvider);
+                await layoutDependencyInstaller.Execute();
+
+                foreach (var viewTemplate in _views)
+                {
+                    var viewName = viewTemplate == "List" ? "Index" : viewTemplate;
+                    // ToDo: This is duplicated from ViewGenerator.
+                    bool isLayoutSelected = controllerGeneratorModel.UseDefaultLayout ||
+                        !String.IsNullOrEmpty(controllerGeneratorModel.LayoutPage);
+
+                    var viewTemplateModel = new ViewGeneratorTemplateModel()
+                    {
+                        ViewDataTypeName = modelTypeAndContextModel.ModelType.FullName,
+                        ViewDataTypeShortName = modelTypeAndContextModel.ModelType.Name,
+                        ViewName = viewName,
+                        LayoutPageFile = controllerGeneratorModel.LayoutPage,
+                        IsLayoutPageSelected = isLayoutSelected,
+                        IsPartialView = false,
+                        ReferenceScriptLibraries = controllerGeneratorModel.ReferenceScriptLibraries,
+                        ModelMetadata = modelTypeAndContextModel.ModelMetadata,
+                        JQueryVersion = "1.10.2"
+                    };
+
+                    // Todo: Need logic for areas
+                    var viewOutputPath = Path.Combine(
+                        ApplicationEnvironment.ApplicationBasePath,
+                        Constants.ViewsFolderName,
+                        controllerRootName,
+                        viewName + Constants.ViewExtension);
+
+                    await CodeGeneratorActionsService.AddFileFromTemplateAsync(viewOutputPath,
+                        viewTemplate + Constants.RazorTemplateExtension, TemplateFolders, viewTemplateModel);
```

Comment:
This actually generates a file for the view. So the flow is Controller generates views, but calls generation directly , not using ViewGenerator at all (which essentially does similar thing. I see that ViewGenerator does more and is called with Alias view (-view parameter?) , it just feels like naturally controller should use ViewGenerator and not know how views are generated .. ? Existing code also looks good, I am just trying to  see if I understand things right...